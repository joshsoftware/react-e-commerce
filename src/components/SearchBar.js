import React from 'react';
import ButtonWrapper from './ButtonWrapper';
import FormInput from './FormInput';
import PropTypes from 'prop-types';
import RowWrapper from './RowWrapper';
import ColumnWrapper from './ColumnWrapper';
const SearchBar = ({ placeholder, setSearch, filterProduct }) => {
  const setSearchWrapper = (e) => {
    setSearch(e.target.value);
  };
  const filterProductWrapper = () => {
    let arr = filterProduct();
    console.log(arr);
  };
  return (
    <>
      {/* <RowWrapper>
           <ColumnWrapper sm={2}>   */}
      <FormInput placeholder={placeholder} onChange={setSearchWrapper} />
      <ButtonWrapper buttonText={'Search'} onClick={filterProductWrapper} />
      {/* </ColumnWrapper>
      </RowWrapper> */}
    </>
  );
};
export default SearchBar;
SearchBar.propTypes = {
  placeholder: PropTypes.string,
  setSearch: PropTypes.func,
  filterProduct: PropTypes.func
};
