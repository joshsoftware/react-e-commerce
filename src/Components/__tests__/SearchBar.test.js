import React from 'react';
import { render } from '@testing-library/react';
import SearchBar from '../SearchBar';

describe('SearchBar ', () => {
  it('Must return entire SearchBar', () => {
    const placeholder = '';
    const setSearch = () => {};
    const filterProduct = () => {};

    const { asFragment } = render(
      <SearchBar placeholder={placeholder} setSearch={setSearch} filterProduct={filterProduct} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
