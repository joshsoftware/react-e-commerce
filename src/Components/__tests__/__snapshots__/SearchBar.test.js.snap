// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar  Must return entire SearchBar 1`] = `
<DocumentFragment>
  <input
    class="form-control"
    placeholder=""
    type="text"
    value=""
  />
  <button
    class="btn btn-secondary"
    type="button"
  >
    Search
  </button>
</DocumentFragment>
`;
